src/Common/gui/PreferencesDialog.cpp

```diff
@@ -51,6 +52,16 @@ void PreferencesDialog::initialize(PreferencesTab initialTab, const Persistence:
     Q_UNUSED(initialTab);
     this->settings = settings;
     this->jamRecorders = jamRecorders;
+    this->jamRecorderCheckBoxes = QMap<QCheckBox *, QString>();
+
+    foreach(const QString jamRecorder, jamRecorders->keys()) {
+        QCheckBox *myCheckBox = new QCheckBox;
```

Comment:
use *this* as parameter in QCheckBox constructor to unsure the CheckBox instance will be deleted in PreferencesDialog destructor (Qt parentship).
``